itest/assets_test.go

```diff
@@ -2200,3 +2221,108 @@ func assertInvoiceState(t *testing.T, hn *HarnessNode, payAddr []byte,
+	)
+	require.NoError(t, err)
+
+	require.NotEmpty(t, closedChannels.Channels)
+
+	targetChanPointStr := fmt.Sprintf("%v:%v", chanPoint.GetFundingTxidStr(),
+		chanPoint.GetOutputIndex())
+
+	var closedChan *lnrpc.ChannelCloseSummary
+	for _, ch := range closedChannels.Channels {
+		if ch.ChannelPoint == targetChanPointStr {
+			closedChan = ch
+			break
+		}
+	}
+	require.NotNil(t, closedChan)
+	require.NotEmpty(t, closedChan.CustomChannelData)
+
+	var closeData rfqmsg.JsonAssetChannel
+	err = json.Unmarshal(closedChan.CustomChannelData, &closeData)
+	require.NoError(t, err)
+
+	require.GreaterOrEqual(t, len(closeData.FundingAssets), 1)
+}
+
+// assertPendingForceCloseChannelAssetData asserts that the pending force close
+// channel has the expected asset data.
+func assertPendingForceCloseChannelAssetData(t *testing.T, node *HarnessNode,
+	chanPoint *lnrpc.ChannelPoint) {
+
+	ctxb := context.Background()
+
+	// Make sure we can find the closed channel in the channel database.
+	pendingChannels, err := node.PendingChannels(
+		ctxb, &lnrpc.PendingChannelsRequest{},
+	)
+	require.NoError(t, err)
+
+	require.NotEmpty(t, pendingChannels.PendingForceClosingChannels)
+
+	targetChanPointStr := fmt.Sprintf("%v:%v", chanPoint.GetFundingTxidStr(),
+		chanPoint.GetOutputIndex())
+
+	var pendingChan *lnrpc.PendingChannelsResponse_PendingChannel
+	for _, ch := range pendingChannels.PendingForceClosingChannels {
+		if ch.Channel.ChannelPoint == targetChanPointStr {
+			pendingChan = ch.Channel
+			break
+		}
+	}
+	require.NotNil(t, pendingChan)
+	require.NotEmpty(t, pendingChan.CustomChannelData)
+
+	var closeData rfqmsg.JsonAssetChannel
+	err = json.Unmarshal(pendingChan.CustomChannelData, &closeData)
+	require.NoError(t, err)
+
+	require.GreaterOrEqual(t, len(closeData.FundingAssets), 1)
+}
+
+// assertWaitingCloseChannelAssetData asserts that the waiting close channel has
+// the expected asset data.
+func assertWaitingCloseChannelAssetData(t *testing.T, node *HarnessNode,
+	chanPoint *lnrpc.ChannelPoint) {
+
+	ctxb := context.Background()
+
+	// Make sure we can find the closed channel in the channel database.
+	pendingChannels, err := node.PendingChannels(
+		ctxb, &lnrpc.PendingChannelsRequest{},
+	)
+	require.NoError(t, err)
+
+	require.NotEmpty(t, pendingChannels.WaitingCloseChannels)
+
+	targetChanPointStr := fmt.Sprintf("%v:%v", chanPoint.GetFundingTxidStr(),
+		chanPoint.GetOutputIndex())
+
+	var waitingCloseChan *lnrpc.PendingChannelsResponse_PendingChannel
+	for _, ch := range pendingChannels.WaitingCloseChannels {
+		if ch.Channel.ChannelPoint == targetChanPointStr {
+			waitingCloseChan = ch.Channel
+			break
+		}
+	}
+	require.NotNil(t, waitingCloseChan)
+	require.NotEmpty(t, waitingCloseChan.CustomChannelData)
+
+	var closeData rfqmsg.JsonAssetChannel
+	err = json.Unmarshal(waitingCloseChan.CustomChannelData, &closeData)
+	require.NoError(t, err)
+
+	require.GreaterOrEqual(t, len(closeData.FundingAssets), 1)
+}
```

Comment:
I tried to come up with a clever way for a helper function, so that these three funcs don't have so much similar code, but I fell short. It's just slightly different at different spots such that 3 separate funcs are warranted. :+1: